cmd/lakectl/cmd/dbt.go

```diff
@@ -110,53 +58,57 @@ var dbtCreateBranchSchema = &cobra.Command{
 		// this could be skipped by using the skip-views flag
 		if !skipViews {
-			dbtClient.ValidateGenerateSchemaMacro()
+			err := ValidateGenerateSchemaMacro(projectRoot, macrosDirName, generateSchemaName, lakectlIdentifier)
+			if err != nil {
+				DieErr(err)
+			}
 		}
 
-		schema := dbtClient.Debug()
+		schema := retrieveSchema(projectRoot)
 		ctx := cmd.Context()
 
 		if shouldCreateBranch {
 			handleBranchCreation(ctx, schema, branchName, metastoreClient)
 		}
 
-		copySchemaWithDbtTables(ctx, continueOnError, continueOnSchemaExists, schema, toSchema, branchName, dbfsLocation, dbtClient, metastoreClient)
+		copySchemaWithDbtTables(ctx, continueOnError, continueOnSchemaExists, schema, toSchema, branchName, dbfsLocation, projectRoot, metastoreClient)
 
 		if skipViews {
 			fmt.Println("skipping views")
 		} else {
-			createViews(dbtClient, toSchema)
+			createViews(projectRoot, toSchema)
 		}
 
-		fmt.Printf("create-branch-schema done! \nyour new schema is %s\nyou can now edit your dbt profile and run dbt on branch %s\n", toSchema, branchName)
+		fmt.Printf("create-branch-schema done!\nyour new schema is %s\nyou can now edit your dbt profile and run dbt on branch %s\n", toSchema, branchName)
 	},
 }
 
-func handleBranchCreation(ctx context.Context, schema, branchName string, metastoreClient metastore.Client) {
-	sourceRepo, sourceBranch, err := ExtractRepoAndBranchFromDBName(ctx, schema, metastoreClient)
-	if err != nil {
-		DieErr(errorNoSourceBranch)
-	}
-	sourceBranchURI, err := GenerateLakeFSURI(sourceRepo, sourceBranch)
+func retrieveSchema(projectRoot string) string {
+	schemaOutput, err := DbtDebug(projectRoot, schemaRegex, commandExecutioner{})
 	if err != nil {
+		fmt.Println(schemaOutput)
 		DieErr(err)
 	}
-	destinationBranchURI, err := GenerateLakeFSURI(sourceRepo, branchName)
+	fmt.Println("dbt debug succeeded with schemaOutput:", schemaOutput)
+	return schemaOutput
+}
+
+func handleBranchCreation(ctx context.Context, schema, branchName string, metastoreClient metastore.Client) {
+	sourceRepo, sourceBranch, err := ExtractRepoAndBranchFromDBName(ctx, schema, metastoreClient)
 	if err != nil {
-		DieErr(err)
+		DieFmt(`Given schema has no source branch associated with it %s`, err)
 	}
-	CreateBranch(ctx, sourceBranchURI, destinationBranchURI)
+	CreateBranch(ctx, sourceRepo, sourceBranch, branchName)
 }
 
 // copySchemaWithDbtTables create a copy of fromSchema and copies all dbt models materialized as table or incremental
-func copySchemaWithDbtTables(ctx context.Context, continueOnError, continueOnSchemaExists bool, fromSchema, toSchema, branchName, dbfsLocation string, dbtClient dbtCommandClient, client metastore.Client) {
+func copySchemaWithDbtTables(ctx context.Context, continueOnError, continueOnSchemaExists bool, fromSchema, toSchema, branchName, dbfsLocation, projectRoot string, client metastore.Client) {
```

Comment:
Another nit if we're already here: style is "DBT" not "Dbt".

```suggestion
func copySchemaWithDBTTables(ctx context.Context, continueOnError, continueOnSchemaExists bool, fromSchema, toSchema, branchName, dbfsLocation, projectRoot string, client metastore.Client) {
```